### PR TITLE
feature: asset info endpoints

### PR DESCRIPTION
**Pull request by NassirAlawar**

### CODE REVIEW

File: lib/Services/cayenne.js [hunk 1]
```diff
@@ -125,6 +125,90 @@ class Cayenne {
...
+    * @param {UUID?} payload.group_id The group_id of the parent group
+    * @param {String} payload.value The value, contents vary based on value_type
+    * @param {String} payload.value_type The type of content contained in value string
+    */
+   createOrUpdateAttribute(device_id, group_id, attribute_id, payload) {
+    var url = null
+    if(attribute_id == null){
+      url = `${this.url}/things/${device_id}/groups/${group_id}`;
+    } else {
+      url = `${this.url}/things/${device_id}/groups/${group_id}/${attribute_id}`;
```
Comment (hugocortes) [inline lib/Services/cayenne.js]: Is `attributes` missing from the path (also on L#204)?  `${this.url}/things/${device_id}/groups/${group_id}/attributes/${attribute_id}`

File: lib/Services/cayenne.js [hunk 2]
```diff
@@ -125,6 +125,90 @@ class Cayenne {
...
+   }
+
+   /**
+    * Delete a group
+    * @param {String} device_id 
+    * @param {String} group_id 
+    */
+   deleteGroup(device_id, group_id) {
+     const url = `/things/${device_id}/groups/${group_id}`;
+     return this.auth.send(this.service, 'DELETE', url, { payload });
```
Comment (hugocortes) [inline lib/Services/cayenne.js]: `{ payload }` should be removed from arg list

File: lib/Services/cayenne.js [hunk 3]
```diff
@@ -125,6 +125,90 @@ class Cayenne {
...
+
+  /**
+    * Delete an attribute
+    * @param {String} device_id 
+    * @param {String} group_id 
+    * @param {String} attribute_id
+    */
+   deleteGroup(device_id, group_id, attribute_id) {
+    const url = `/things/${device_id}/groups/${group_id}/${attribute_id}`;
+    return this.auth.send(this.service, 'DELETE', url, { payload });
```
Comment (hugocortes) [inline lib/Services/cayenne.js]: `{ payload }` should be removed from arg list

File: lib/Services/cayenne.js [hunk 4]
```diff
@@ -125,6 +125,90 @@ class Cayenne {
...
 
+  /*
+   * Begin Device Attribute Routes
+   */
+
+   /**
+    * Gets all attribute groups and attributes for a device
+    * @param {String} device_id The device id to get attributes for
+    */
+   getAttributeGroups(device_id) {
```
Comment (hugocortes) [inline lib/Services/cayenne.js]: Query object should be in parameter list with `user_id` key

File: lib/Services/cayenne.js (hunk 1 above)
Comment (NassirAlawar) [reply lib/Services/cayenne.js]: yup :D

**Pull request merged**